app/src/lib/progress/from-process.ts

```diff
@@ -62,9 +63,17 @@ function createProgressProcessCallback(
 
       process.on('close', () => {
         disposable.dispose()
-        // NB: We don't really care about errors deleting the file, but Node
-        // gets kinda bothered if we don't provide a callback.
-        Fs.unlink(lfsProgressPath, () => {})
+        // the order of these callbacks is important because
+        //  - unlink can only be done on files
+        //  - rmdir can only be done when the directory is empty
+        //  - we don't want to surface errors to the user if something goes
+        //    wrong (these files can stay in TEMP and be cleaned up eventually)
+        Fs.unlink(lfsProgressPath, err => {
+          if (err == null) {
+            const directory = Path.dirname(lfsProgressPath)
+            Fs.rmdir(directory, () => {})
```

Comment:
Is it worth capturing any potential errors when calling `FS.rmdir`? At the moment, they get tossed out.

Reply:
I don't think it's worth capturing and logging, for two reasons:

 - there's not really much the user can do at this point - we need to rely on the OS to handle these temporary files if we can't clean them up
 - logging this error out might be a minor help for the user, but as this can potentially occur whenever we spawn Git LFS (so fetch/push/pull) it could lead to a lot of noisy logs